### DISCUSSION TITLE
Streamlit based Web App for ASR using Whisper Models

### DISCUSSION THREAD

**Discussion by prateekralhan** [openai/whisper#148] [Show and tell]
Hi,

Kudos to the team for their work on ASR. I worked on developing a simple streamlit based web-app on automatic speech recognition for different audio formats using OpenAI's Whisper models 😄!

Title: Automatic Speech Recognition using OpenAI's Whisper
Github Repo: https://github.com/prateekralhan/OpenAI_Whisper_ASR

![192354340-7a242d6a-8d71-4b91-ae29-c35fbafe3ad4](https://user-images.githubusercontent.com/29462447/192449607-bd638742-b32d-4669-bcdd-f55fd0e4a9b7.gif)

Happy opensourcing!

**Comment by Abdurrafey-Siddiqui** [comment]
Hi can you add options that allow us to tweak the transcribe like temperature, intitial_prompt ect..

**Reply by prateekralhan** [reply]
@Abdurrafey-Siddiqui  currently I don't have the bandwidth to work on this. If you wish. you can raise a PR with these refinements else give me some time please to work on this.

**Comment by gustavz** [comment]
It would be nice to be able to record audio via browser. Unfortunately I did not find a proper solution for that using streamlit.
Using system audio of course works, but for a deployed app only client side audio makes sense.
Do you have a solution for that?

**Reply by prateekralhan** [reply]
@gustavz , can you have a look at this?

* https://github.com/whitphx/streamlit-webrtc

**Reply by gustavz** [reply]
yeah that's what I ended up using. Check it out the project: https://github.com/gustavz/jaivus

**Reply by prateekralhan** [reply]
This looks pretty cool. Great stuff!
